### PR TITLE
Create index.html and add navbar and hero section

### PR DESCRIPTION
**Pull request by Jojokats**
I just created a quick navbar and hero section for the index.html, something quick just to get started. Let me know if this is OK Richard.

hero

### CODE REVIEW

File: app/npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,45 @@
+0 info it worked if it ends with ok
```
Comment (RichardLitt) [inline app/npm-debug.log]: This file shouldn't be checked in. Can you add it to the `.gitignore`, and remove it in another commit?

File: app/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "bs4starter",
```
Comment (RichardLitt) [inline app/package.json]: What made you go with the bs4starter?

File: app/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+node_modules
+DS_Store
```
Comment (RichardLitt) [inline app/.gitignore]: This should be `.DS_Store`. You'll need to add that, and remove the DS_Store you've added.

**Comment by RichardLitt** [comment]
Let's work a bit more at this PR. What does `Homepage` in the title mean? 

Recommended reading: [Writing good commit messages](https://github.com/erlang/otp/wiki/Writing-good-commit-messages), [writing good PRs](https://github.com/blog/1943-how-to-write-the-perfect-pull-request). 

Try editing the content above after reading these.

File: app/npm-debug.log (hunk 1 above)
Comment (Jojokats) [reply app/npm-debug.log]: Ok, will add it ``` .gitignore```

File: app/package.json (hunk 2 above)
Comment (Jojokats) [reply app/package.json]: I use this to start my projects

File: app/.gitignore (hunk 3 above)
Comment (Jojokats) [reply app/.gitignore]: Done

Comment (RichardLitt) [reply app/.gitignore]: Still seeing it?

**Comment by Jojokats** [comment]
>Try editing the content above after reading these.

Not sure I understand which content you are referring to.

**Comment by RichardLitt** [comment]
The title and the first comment in this PR. Your comment is fine (well done!), but the title could be improved.

**Comment by Jojokats** [comment]
>What was the decision behind putting everything in an app folder? Why not in the base folder?

I could change the name to base

**Comment by RichardLitt** [comment]
Base is another word for root directory: https://en.wikipedia.org/wiki/Root_directory. Here, I mean the main directory where the README is.

File: app/.gitignore (hunk 3 above)
Comment (Jojokats) [reply app/.gitignore]: Done locally. I haven't pushed yet, will do shortly

**Comment by Jojokats** [comment]
>The title and the first comment in this PR. Your comment is fine (well done!), but the title could be improved.

If I understand correctly, you want me to change the title ***Homepage*** to a better title?

**Comment by RichardLitt** [comment]
Yep!

**Comment by Jojokats** [comment]
Is this ok?
>Our first index.html page for The Travel Shelf.

**Comment by RichardLitt** [comment]
Close. Take a look at [this article](https://github.com/erlang/otp/wiki/Writing-good-commit-messages). 

Then look at some PRs for popular projects:
- https://github.com/reactjs/redux/pulls
- https://github.com/angular/angular/pulls
- https://github.com/atom/atom/pulls

**Comment by Jojokats** [comment]
Is this better?

>First navbar | Hero section in index.html for The Travel Shelf.

**Comment by RichardLitt** [comment]
Take a look at the PRs in redux and atom - how similar is your request?  What would you change to make it look like one of those PRs?

**Comment by Jojokats** [comment]
How about this:

>Added navbar | Hero section in index.html.

**Comment by RichardLitt** [comment]
Close! 

What else does this do? It creates an index.html, right? Let's make that explicit in the title.

**Comment by Jojokats** [comment]
```
git status
On branch homepage
Your branch is up-to-date with 'origin/homepage'.
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	deleted:    ../app/.DS_Store
	deleted:    ../app/.gitignore
	deleted:    ../app/README.md
	deleted:    ../app/gulpfile.js
	deleted:    ../app/npm-debug.log
	deleted:    ../app/package.json
	deleted:    ../app/src/.DS_Store
	deleted:    ../app/src/css/bootstrap.css
	deleted:    ../app/src/css/font-awesome.min.css
	deleted:    ../app/src/css/style.css
	deleted:    ../app/src/fonts/FontAwesome.otf
	deleted:    ../app/src/fonts/fontawesome-webfont.eot
	deleted:    ../app/src/fonts/fontawesome-webfont.svg
	deleted:    ../app/src/fonts/fontawesome-webfont.ttf
	deleted:    ../app/src/fonts/fontawesome-webfont.woff
	deleted:    ../app/src/fonts/fontawesome-webfont.woff2
	deleted:    ../app/src/img/annie.jpg
	deleted:    ../app/src/img/ts-logo.jpg
	deleted:    ../app/src/img/ts-logo.svg
	deleted:    ../app/src/img/ts-logo@2x.jpg
	deleted:    ../app/src/img/ts-logo@3x.jpg
	deleted:    ../app/src/index.html
	deleted:    ../app/src/js/bootstrap.min.js
	deleted:    ../app/src/js/jquery.min.js
	deleted:    ../app/src/js/navbar-fixed.js
	deleted:    ../app/src/js/popper.min.js
	deleted:    ../app/src/scss/style.scss

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	../.DS_Store
	./
	../npm-debug.log
	../travel-shelf.sketch

no changes added to commit (use "git add" and/or "git commit -a")
Joannas-MacBook-Pro:app joannakatsifaraki$ 

```

**Comment by RichardLitt** [comment]
That sounds good to me! Change it to that. No need to capitalize navbar and hero.

**Comment by Jojokats** [comment]
OK, so I will change ```homepage``` to:

Created index.html and added navbar and hero section.

**Comment by Jojokats** [comment]
OK, I've never done this before, so I googled it and got these instructions:
* Rename your local branch. If you are on the branch you want to rename: git branch -m new-name. ...
* Delete the old-name remote branch and push the new-name local branch. git push origin :old-name new-name.
* Reset the upstream branch for the new-name local branch. Switch to the branch and then: git push origin -u new-name.

**Comment by RichardLitt** [comment]
> OK, I've never done this before, so I googled it and got these instructions:

You shouldn't have done any of this before! :D Welcome. 

No, don't change the branch name! Just change the title of this PR. Click the edit button: 

screen shot 2017-10-04 at 11 20 30 am

File: app/.gitignore (hunk 3 above)
Comment (RichardLitt) [reply app/.gitignore]: Cool. In the future, if it hasn't been pushed and solved in this PR, then it isn't done.

**Comment by Jojokats** [comment]
Done! few.....

**Comment by RichardLitt** [comment]
Super, super close. Generally, commits and PRs use the active tense. Create, over Created.

**Comment by Jojokats** [comment]
Yes, I saw that, but find that weird, cause I already created it, and ```create``` is giving an order to do this, when it's already done!

**Comment by RichardLitt** [comment]
It's not already done. It hasn't been merged into yet. It's also just a standard we use in the industry.

**Comment by Jojokats** [comment]
How do I merge, via the command line?

**Comment by RichardLitt** [comment]
It's best practice on a shared repository to not merge your own pull requests. Merging means that a PR is good - which you can't know for yourself. 

There's still work to do here. Move the app folder and commit it.

**Comment by Jojokats** [comment]
I moved the files to the root as requested but am getting problems with git:
```
Joannas-MacBook-Pro:app joannakatsifaraki$ git status
On branch homepage
Your branch is up-to-date with 'origin/homepage'.
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   .DS_Store
	modified:   .gitignore
	deleted:    src/.DS_Store
	deleted:    src/css/bootstrap.css
	deleted:    src/css/font-awesome.min.css
	deleted:    src/css/style.css
	deleted:    src/fonts/FontAwesome.otf
	deleted:    src/fonts/fontawesome-webfont.eot
	deleted:    src/fonts/fontawesome-webfont.svg
	deleted:    src/fonts/fontawesome-webfont.ttf
	deleted:    src/fonts/fontawesome-webfont.woff
	deleted:    src/fonts/fontawesome-webfont.woff2
	deleted:    src/img/annie.jpg
	deleted:    src/img/ts-logo.jpg
	deleted:    src/img/ts-logo.svg
	deleted:    src/img/ts-logo@2x.jpg
	deleted:    src/img/ts-logo@3x.jpg
	deleted:    src/index.html
	deleted:    src/js/bootstrap.min.js
	deleted:    src/js/jquery.min.js
	deleted:    src/js/navbar-fixed.js
	deleted:    src/js/popper.min.js
	deleted:    src/scss/style.scss

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	../.DS_Store
	css/
	fonts/
	img/
	index.html
	js/
	scss/
	../npm-debug.log
	../travel-shelf.sketch

no changes added to commit (use "git add" and/or "git commit -a")
Joannas-MacBook-Pro:app joannakatsifaraki$ 

```

**Comment by RichardLitt** [comment]
Let's break it down. 

What do you expect to happen? 

What actually happened?

**Comment by RichardLitt** [comment]
This pull request will close #32.

**Comment by Jojokats** [comment]
Sorry for the delay @RichardLitt I had to step out and am back now :) 
>Let's break it down.
What do you expect to happen?
What actually happened?

I removed the src folder and git is thinking I deleted everything in the folder?
But it also is showing me untracked files
```
        css/
	fonts/
	img/
	index.html
	js/
	scss/
```
These are the folders I moved out of the src folder

**Comment by RichardLitt** [comment]
Try `git add .`. What happens?

**Comment by Jojokats** [comment]
So I did git add . and I got this:
```
Joannas-MacBook-Pro:app joannakatsifaraki$ git add .
Joannas-MacBook-Pro:app joannakatsifaraki$ git status
On branch homepage
Your branch is up-to-date with 'origin/homepage'.
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	modified:   .DS_Store
	modified:   .gitignore
	renamed:    src/css/bootstrap.css -> css/bootstrap.css
	renamed:    src/css/font-awesome.min.css -> css/font-awesome.min.css
	renamed:    src/css/style.css -> css/style.css
	renamed:    src/fonts/FontAwesome.otf -> fonts/FontAwesome.otf
	renamed:    src/fonts/fontawesome-webfont.eot -> fonts/fontawesome-webfont.eot
	renamed:    src/fonts/fontawesome-webfont.svg -> fonts/fontawesome-webfont.svg
	renamed:    src/fonts/fontawesome-webfont.ttf -> fonts/fontawesome-webfont.ttf
	renamed:    src/fonts/fontawesome-webfont.woff -> fonts/fontawesome-webfont.woff
	renamed:    src/fonts/fontawesome-webfont.woff2 -> fonts/fontawesome-webfont.woff2
	renamed:    src/img/annie.jpg -> img/annie.jpg
	renamed:    src/img/ts-logo.jpg -> img/ts-logo.jpg
	renamed:    src/img/ts-logo.svg -> img/ts-logo.svg
	renamed:    src/img/ts-logo@2x.jpg -> img/ts-logo@2x.jpg
	renamed:    src/img/ts-logo@3x.jpg -> img/ts-logo@3x.jpg
	renamed:    src/index.html -> index.html
	renamed:    src/js/bootstrap.min.js -> js/bootstrap.min.js
	renamed:    src/js/jquery.min.js -> js/jquery.min.js
	renamed:    src/js/navbar-fixed.js -> js/navbar-fixed.js
	renamed:    src/js/popper.min.js -> js/popper.min.js
	renamed:    src/scss/style.scss -> scss/style.scss
	deleted:    src/.DS_Store

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	../.DS_Store
	../npm-debug.log
	../travel-shelf.sketch

Joannas-MacBook-Pro:app joannakatsifaraki$ 
```

**Comment by Jojokats** [comment]
So I can commit -m now

**Comment by RichardLitt** [comment]
Remove the .DS_Store file from the index, and make sure it is deleted locally and not tracked in Git. otherwise, looks good. Git can track renames, but if the new location isn't added, it thinks that any moved files are deleted, which is what was happening.

**Comment by Jojokats** [comment]
OK, just deleted .DS_Store and made the commit. Not sure what to do now.
Do I:
git push origin master
or
git merge origin/master

**Comment by RichardLitt** [comment]
Why would you merge? Why would you push? (These are not rhetorical - what are you trying to achieve?)

**Comment by Jojokats** [comment]
OK, so I should just 
git push?

**Comment by RichardLitt** [comment]
I'm sorry, I may have not been clear. Please answer these questions: Why would you merge? Why would you push? These are not rhetorical - what are you trying to achieve? Why would you use one, over the other? What do `git push`, `git push origin master`, and `git merge origin/master` do? Each of them? 

If you're unsure, I would suggest googling that command, or doing a longer git tutorial. Then answer the questions above. Writing the answer should make it clear which to do. 

Technically, I can tell you, but then you won't learn much.

**Comment by Jojokats** [comment]
OK will do the research

**Comment by Jojokats** [comment]
>What do git push, git push origin master, and git merge origin/master do? Each of them?

git push:
- will push any commits from local repository to the remote repository & the branch
If I'm on the branch ex: homepage, I can just run: git push.
If I'm on master, I run: git push origin master
Use git push to push commits made on my local branch to a remote repository.

git push origin master:
- Push master branch to origin server
pushes from the local system to the remote repository the master branch

git merge origin:
The git merge command lets you take the independent lines of development created by git branch and integrate them into a single branch.

git merge master:
fast-forward merge - Git places all the commits on the master branch as if we never branched away. It works as long as we don't do any work on master branch before doing the merge.

---------------------------------------------------------------------------

In my case, After I made the changes, I will just run:
git push
(because I'm already in the homepage branch, I don't need to specify a name)

**Comment by Jojokats** [comment]
Thanks so much for your direction, @RichardLitt.  I believe I did the changes you requested and so will re-open this pull request.

**Comment by RichardLitt** [comment]
I'm still seeing an `app` folder, and files in it. Can you move those files to the base directory, and not use an `app` folder?

**Comment by RichardLitt** [comment]
@Jojokats Any word on this?

**Comment by Jojokats** [comment]
@RichardLitt I'm working on it, 👍

**Comment by Jojokats** [comment]
Hello @RichardLitt , I resolved some merge conflicts and removed the app folder. I've noticed that when I do "npm start", a "src/" folder gets created with some default js files underneath, and the server states that it's serving from the src directory. (It tries to serve "src/index.html" and cannot find it.)  It sounds like we should move our code base under an "src/" directory, don't you think?

**Comment by Jojokats** [comment]
@RichardLitt , but if you feel this is the way to do this, feel free to merge in this pull request right now.  Thanks.

**Comment by RichardLitt** [comment]
Thanks! This is getting closer. 

I don't think moving things to the `src` folder makes sense. Why not? Because that is how [`gulp`](https://gulpjs.com/) works. 

Some questions for you to answer to help you understand this:
- What does `npm start` do? Why? Where is it defined?
- What is gulp?
- What is the src folder normally for? 

The package.json file you have in this commit overwrites some things we need. Can you post here which fields shouldn't be changed or removed?

**Comment by RichardLitt** [comment]
> @RichardLitt , but if you feel this is the way to do this, feel free to merge in this pull request right now. Thanks.

Thanks! I'll merge it when it is ready. There's no need to say this in the future. :)

**Comment by Jojokats** [comment]
> Some questions for you to answer to help you understand this:
What does npm start do? Why? Where is it defined?
What is gulp?
What is the src folder normally for?

Ok, will look into this, thanks!

**Comment by Jojokats** [comment]
>What does npm start do

When you issue the com­mand npm start from the root direc­tory of your nodejs project, node will look for a scripts object in your package.json file. If found, it will look for a script with the key start and run the com­mand spec­i­fied as its value.

>What is gulp?

Gulp is a javascript task runner that lets you automate tasks such as…

Bundling and minifying libraries and stylesheets.
Refreshing your browser when you save a file.
Quickly running unit tests
Running code analysis
Less/Sass to CSS compilation
Copying modified files to an output directory

>What is the src folder normally for?

src/ stands for source, and is the raw code before minification or concatenation or some other compilation - used to read/edit the code.

### Also good to know about dist:
dist/ stands for distribution, and is the minified/concatenated version - actually used on production sites.

**Comment by Jojokats** [comment]
I deleted the src directory, and my branch is up to date with origin/homepage.

**Comment by RichardLitt** [comment]
@VernL @ZJaf What do you think about this? I think it looks fine.

**Comment by VernL** [comment]
The page looks great, I like the style. 

I am unfamiliar with gulp. I took the time to look at the docs, it looks useful, however I am unsure how to incorporate it into our existing work. 

It looks to me that the app is now being served by gulpfile.js instead of index.js? Why is that?

**Comment by RichardLitt** [comment]
@ZJaf If you like it, want to go ahead and merge? 

I want to get all of you in the habit of reviewing and merging other's PRs. :)

**Comment by RichardLitt** [comment]
Alright! Merged! Well done, @Jojokats!

**Comment by Jojokats** [comment]
Thanks @RichardLitt !

**Pull request merged**